src/util-logopenfile.c

```diff
@@ -424,6 +438,18 @@ SCConfLogOpenGeneric(ConfNode *conf,
         }
     }
 
+    /* Rotate log file based on size */
+    const char *rotate_size = ConfNodeLookupChildValue(conf, "rotate-size");
+    if (rotate_size != NULL) {
+        log_ctx->flags |= LOGFILE_ROTATE_SIZE;
```

Comment:
Better to have it after the parsing of the option in case we later consider we do not need to exit in case of failure.

Reply:
Would it make sense to just report the error and remove the exit in case of failure in this patch instead of waiting with this for later?

After you mentioned it, I don't think this is something that should make exit Suricata.

Reply:
In #3123 I tried changing it so that it doesn't set the flag if it fails instead of exiting. I also changed 'rotate-interval' so it behaves in the same way.